tests/contrib/vertica/test_vertica.py

```diff
@@ -0,0 +1,341 @@
+# stdlib
+
+# 3p
+import wrapt
+
+# project
+from ddtrace import Pin
+from ddtrace.contrib.vertica.patch import patch, unpatch
+from ddtrace.ext import errors
+
+# testing
+import pytest
+from tests.contrib.config import VERTICA_CONFIG
+from tests.opentracer.utils import init_tracer
+from tests.test_tracer import get_dummy_tracer
+
+from .fixtures import test_conn, test_tracer
+from .utils import override_config
+
+
+TEST_TABLE = "test_table"
+
+
+class TestVerticaPatching(object):
+    def test_not_patched(self):
+        """Ensure that vertica is not patched somewhere before our tests"""
+        import vertica_python
+
+        assert not isinstance(vertica_python.Connection.cursor, wrapt.ObjectProxy)
+        assert not isinstance(
+            vertica_python.vertica.cursor.Cursor.execute, wrapt.ObjectProxy
+        )
+
+    def test_patch_after_import(self):
+        """Patching _after_ the import will not work because we hook into
+        the module import system.
+        """
+        import vertica_python
+
+        patch()
+
+        # use a patched method from each class as indicators
+        assert isinstance(vertica_python.Connection.cursor, wrapt.ObjectProxy)
+        assert isinstance(
+            vertica_python.vertica.cursor.Cursor.execute, wrapt.ObjectProxy
```

Comment:
Shouldn't this be failing?

Reply:
You're right! I forgot to circle back to this. There is some weirdness going on here.